api/buf.lock

```diff
@@ -0,0 +1,9 @@
+# Generated by buf. DO NOT EDIT.
```

Comment:
I presume we don't need this file in SCM?

Reply:
buf.lock is required for [ensuring dependency versions](https://buf.build/docs/bsr/module/dependency-management), seems to be similar with `package-lock.json` or `go.sum`